app/services/subdomain_creator.rb

```diff
@@ -0,0 +1,71 @@
+require_relative 'application_service'
+require 'faraday'
+require 'JSON'
+
+class SubdomainCreator < ApplicationService
+
+    def initialize(subdomain)
+        @base_url = "https://api.heroku.com/apps"
+        @headers = headers
+        @subdomain = subdomain.downcase
+    end
+
+    def headers
+        {
+            Accept: "application/vnd.heroku+json; version=3",
+            Content_Type: "application/json",
+            Authorization: "Bearer 998aa496-ac7a-4b4a-bb35-7d45f74ca343",
+            Accept_encoding: "utf8"
+        }
+    end
+
+    # reload the partner because if the auth fails, the token get renew
+    def call
+        data = payload.body
+        if subdomain_exists?(JSON.parse(data), @subdomain)
+            puts "Subdomain already exists"
+        else
+            create_subdomain
+        end
+    end
+
+    def create_subdomain
+        body = {
+            "hostname": "#{@subdomain}.avenist.com",
+            "sni_endpoint": nil
+        }
+        body = body.to_json
+        Faraday.post("#{@base_url}/equinox-admin/domains", body, @headers)
+    end
+
+    def subdomain_exists?(data, subdomain)
+        subdomain = false
```

Comment:
do you think this subdomain might get mixed up with the parameter "subdomain"?

Reply:
It is the parameter subdomain being passed int the controller so I think it's okay that it is called the same. 👍